### PR TITLE
i18n(es): Translate internationalization.mdx

### PR DESCRIPTION
**Pull request by ansango**
#### Description (required)

Translate internationalization.mdx into -es

### CODE REVIEW

File: src/content/docs/es/guides/internationalization.mdx [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+description: Aprende a usar el enrutamiento i18n en Astro.
+i18nReady: true
+---
+
+import { FileTree } from '@astrojs/starlight/components';
+import Since from '~/components/Since.astro'
+
+La internacionalización (i18n) de Astro te permite adaptar tu proyecto para una audiencia internacional. Esta API de enrutamiento te ayuda a generar, usar y verificar las URL que produce tu sitio multilingüe.
+
+Astro's i18n routing allows you to bring your multilingual content with support for configuring a default language, computing relative page URLs, and accepting preferred languages provided by your visitor's browser. You can also specify fallback languages on a per-language basis so that your visitors can always be directed to existing content on your site.
```
[the remark this answers is not in the document]
Comment (ansango) [reply src/content/docs/es/guides/internationalization.mdx]: Listo 😄

File: src/content/docs/es/guides/internationalization.mdx [hunk 2]
```diff
@@ -0,0 +1,372 @@
+---
+title: Internationalización (i18n) Enrutamiento
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:2]:
```suggestion
title: Enrutamiento con Internationalización (i18n)
```

File: src/content/docs/es/guides/internationalization.mdx [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+Las rutas de i18n de Astro te permiten llevar tu contenido multilingüe con soporte para configurar un idioma predeterminado, calcular las URL de las páginas relativas y aceptar los idiomas preferidos proporcionados por el navegador de tus usuarios. También puedes especificar idiomas por defecto en función de cada idioma para que tus usuarios siempre puedan ser dirigidos un determinado tipo de contenido existente en tu sitio.
+
+## Lógica de enrutamiento
+
+Astro utiliza un [middleware](/es/guides/middleware/) para implementar su lógica de enrutamiento. Esta función de middleware se coloca en la [primera posición](/es/guides/middleware/#encadenando-el-middleware) donde espera cada `Response` que proviene de cualquier middleware adicional y de cada ruta antes de ejecutar finalmente su propia lógica.
+
+Estro significa que las operaciones (por ejemplo, redirecciones) de tu propio middleware y tu lógica de página se ejecutan primero, tus rutas se renderizan y luego el middleware de i18n realiza sus propias acciones, como verificar que una URL localizada corresponda a una ruta válida.
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:18]:
```suggestion
Esto significa que las operaciones (por ejemplo, redirecciones) de tu propio middleware y tu lógica de página se ejecutan primero, tus rutas se renderizan y luego el middleware de i18n realiza sus propias acciones, como verificar que una URL localizada corresponda a una ruta válida.
```

File: src/content/docs/es/guides/internationalization.mdx [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+      - index.astro
+</FileTree>
+
+:::note
+Las carpetas localizadas no necesitan estar en la raíz de la carpeta `/pages/`.
+:::
+
+### Crear enlaces
+
+With i18n routing configured, you can now compute links to pages within your site using the helper functions such as [`getRelativeLocaleURL()`](/en/reference/api-reference/#getrelativelocaleurl) available from the [`astro:i18n` module](/en/reference/api-reference/#internationalization-astroi18n). These generated links will always provide the correct, localized route and can help you correctly use, or check, URLs on your site.
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:62]:
```suggestion
```

File: src/content/docs/es/guides/internationalization.mdx [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+Con el enrutamiento i18n configurado, ahora puedes calcular enlaces a páginas dentro de tu sitio utilizando las funciones de ayuda como [`getRelativeLocaleURL()`](/es/reference/api-reference/#getrelativelocaleurl) disponibles en el módulo [`astro:i18n`](es/reference/api-reference/#internationalization-astroi18n). Estos enlaces generados siempre proporcionarán la ruta localizada correcta y pueden ayudarte a usar o verificar correctamente las URL en tu sitio.
+
+
+También puedes seguir escribiendo los enlaces manualmente.
+
+```astro title="src/pages/es/index.astro"
+---
+import { getRelativeLocaleUrl } from 'astro:i18n';
+
+// defaultLocale is "es"
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:73]:
```suggestion
// defaultLocale es "es"
```

File: src/content/docs/es/guides/internationalization.mdx [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+  }
+})
+```
+
+Astro proporciona funciones de ayuda para tu middleware para que puedas controlar tu propio enrutamiento predeterminado, excepciones, comportamiento de fallback, captura de errores, etc: [`redirectToDefaultLocale()`](/es/reference/api-reference/#redirecttodefaultlocale), [`notFound()`](/es/reference/api-reference/#notfound) y [`redirectToFallback()`](/es/reference/api-reference/#redirecttofallback):
+
+
+```js title="src/middleware.js"
+import { defineMiddleware } from "astro:middleware";
+import { redirectToDefaultLocale } from "astro:i18n"; // función disponible con `manual` routing
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:206]:
```suggestion
import { redirectToDefaultLocale } from "astro:i18n"; // función disponible con enrutamiento `manual`
```

File: src/content/docs/es/guides/internationalization.mdx [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+        codes: ["fr", "fr-BR", "fr-CA"]
+    }],
+    routing: {
+        prefixDefaultLocale: true
+    }
+  }
+})
+```
+
+Cuando uses funciones del módulo virtual [`astro:i18n`](/es/reference/api-reference/#internacionalización-astroi18n) para calcular rutas de URL válidas basadas en tu configuración (por ejemplo, `getRelativeLocaleUrl()`), [usa el `path` como valor para `locale`](/es/reference/api-reference/#getlocalebypath).
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:274]:
```suggestion
Cuando uses funciones del [módulo virtual `astro:i18n`](/es/reference/api-reference/#internacionalización-astroi18n) para calcular rutas de URL válidas basadas en tu configuración (por ejemplo, `getRelativeLocaleUrl()`), [usa el `path` como valor para `locale`](/es/reference/api-reference/#getlocalebypath).
```

File: src/content/docs/es/guides/internationalization.mdx [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+
+- [`Astro.preferredLocaleList`](/es/reference/api-reference/#astropreferredlocalelist): Un array de todos los idiomas que son solicitados por el navegador y admitidos por tu sitio web. Esto produce una lista de todos los idiomas compatibles entre tu sitio y tu visitante. El valor es `[]` si ninguno de los idiomas solicitados por el navegador se encuentra en tu matriz `locales`. Si el navegador no especifica ningún idioma preferido, entonces este valor será [`i18n.locales`].
+
+- [`Astro.currentLocale`](/es/reference/api-reference/#astrocurrentlocale): El idioma calculado a partir de la URL actual, utilizando la sintaxis especificada en tu configuración de `locales`. Si la URL no contiene un prefijo `/[locale]/`, entonces el valor se establecerá en [`i18n.defaultLocale`](/es/reference/configuration-reference/#i18ndefaultlocale).
+
+Para que coincidan correctamente con las preferencias de tus usuarios, proporciona tus `codes` utilizando el mismo patrón [utilizado por el navegador](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language#syntax).
+
+## Estrategia de respaldo o Fallback
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:290]:
```suggestion
## Estrategia de respaldo
```

File: src/content/docs/es/guides/internationalization.mdx [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+
+Las URL anteriores también serán devueltas por las funciones `getAbsoluteLocaleUrl()` y `getAbsoluteLocaleUrlList()`.
+
+#### Limitaciones
+
+
+Esta función tiene algunas restricciones:
+
+- La opción `site` es obligatoria.
+- El `output` debe ser `server` para que esta función funcione.
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:360]:
```suggestion
- El `output` debe ser `server`.
```

File: src/content/docs/es/guides/internationalization.mdx [hunk 10]
```diff
@@ -0,0 +1,372 @@
...
+- No puede haber páginas prerenderizadas individuales.
+- The adapter feature [`functionPerRoute`](/en/reference/adapter-reference/#functionperroute) is not supported.
+- La funcionalidad del adaptador [`functionPerRoute`](/es/reference/adapter-reference/#functionperroute) no es compatible.
+
+Astro depende de los siguientes encabezados para admitir la función:
+
+- [`X-Forwarded-Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host) y [`Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host). Astro usará lo principal primero. Si no existe, intentará lo secundario.
+- [`X-Forwarded-Proto`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) y [`URL#protocol`](https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol) de la solicitud.
```
Comment (Waxer59) [inline src/content/docs/es/guides/internationalization.mdx:368]:
```suggestion
- [`X-Forwarded-Proto`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) y [`URL#protocol`](https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol) de la solicitud del servidor.
```

**Review by Waxer59 — COMMENTED** [review]
Buena PR 🤠, te dejo unas sugerencias!

**Review by Waxer59 — APPROVED** [review]

File: src/content/docs/es/guides/internationalization.mdx [hunk 11]
```diff
@@ -0,0 +1,371 @@
...
+
+#### Limitaciones
+
+
+Esta función tiene algunas restricciones:
+
+- La opción `site` es obligatoria.
+- El `output` debe ser `server`.
+- No puede haber páginas prerenderizadas individuales.
+- The adapter feature [`functionPerRoute`](/en/reference/adapter-reference/#functionperroute) is not supported.
```
Comment (dreyfus92) [inline src/content/docs/es/guides/internationalization.mdx:361]:
```suggestion
```

**Review by dreyfus92 — COMMENTED** [review]
Buen trabajo @ansango 😁, te dejo una pequeña sugerencia.

**Comment by ansango** [comment]
> Buen trabajo @ansango 😁, te dejo una pequeña sugerencia.

Gracias, 8 ojos ven más que dos! 😅

**Review by dreyfus92 — APPROVED** [review]
Muchas gracias @ansango y bienvenido a `astro-docs` 🫡

**Comment by dreyfus92** [comment]
@ansango, parece que todavía queda pendiente resolver los links que están mal. Si necesitas ayuda me puedes pinguear aquí o en discord.

**Comment by ansango** [comment]
@dreyfus92 Creo que ya deberia de funcionar todo correctamente, links arreglados y probados 😄

**Comment by yanthomasdev** [comment]
Looks like we're missing a few headings in the Spanish translation of API Reference (e.g. [Astro.preferredLocale](https://docs.astro.build/en/reference/api-reference/#astropreferredlocale)), that's why the broken links. Could any of you take a look and make sure to add these to the `api-reference.mdx` translation?

**Comment by ansango** [comment]
> Looks like we're missing a few headings in the Spanish translation of API Reference (e.g. [Astro.preferredLocale](https://docs.astro.build/en/reference/api-reference/#astropreferredlocale)), that's why the broken links. Could any of you take a look and make sure to add these to the `api-reference.mdx` translation?

Thank you for the review @yanthomasdev, I fixed and added new translations in `api-reference.mdx`. I hope build will be ok! 😄

**Comment by dreyfus92** [comment]
There are still 4 broken links, we're almost there to ship this @ansango 😁

**Comment by ansango** [comment]
> There are still 4 broken links, we're almost there to ship this @ansango 😁

Sure! I reviewed the previous gh action, and I built on my desktop, so I hope that we'll be ready now 😄

**Comment by dreyfus92** [comment]
Muchas gracias @ansango, ya quedó 🥳

**Pull request merged**